runtime/major_gc.c

```diff
@@ -620,7 +625,9 @@ static void mark_slice (intnat work)
       }
     } else if (caml_gc_subphase == Subphase_mark_roots) {
       CAML_EV_BEGIN(EV_MAJOR_MARK_ROOTS);
+      marked_words -= work;
       work = caml_darken_all_roots_slice (work);
```

Comment:
There is a very slight inaccuracy here: `caml_darken_all_roots_slice` counts 1 unit of work for each root, regardless of whether or not it was actually marked. (This probably doesn't make any difference in practice, though)

Reply:
Well it's inaccurate anyway because the roots are not taken into account in the slice size computation, so this is, in a sense, extra work.